runtime/doc/lsp.txt

```diff
@@ -2179,4 +2179,34 @@ resolve_capabilities({server_capabilities})
     Return: ~
         (table) Normalized table of capabilities
 
+
+==============================================================================
+Lua module: vim.lsp.inlay_hint                                *lsp-inlay_hint*
+
+clear({client_id}, {bufnr})                       *vim.lsp.inlay_hint.clear()*
```

Comment:
"inlay hints" seems fairly special-case for a dedicated module (from a user interface perspective). Does this need to be exposed as a different module or could/should it live in `vim.lsp.buf` ? 

Does any of this actually need to be public yet? Doesn't the server provide the hints, so we don't need this public interface?

Reply:
The server provides the hints when you request for them. It needs to be public for now as there is no auto refresh mechanism yet - after this gets merged, I'll make a follow up PR for the auto refresh mechanism which will add a toggle function in vim.lsp.but

(See the discussion in this PR with clason and Mathias)

Reply:
> after this gets merged, I'll make a follow up PR for the auto refresh mechanism which will add a toggle function in vim.lsp.but

But at least the `inlay_hints` module should stay private until we know what parts actually need to be public. That means prefix its name with underscore, and it should not have docs in `:help lsp`.

Reply:
done